### PR TITLE
CircleCI: dockerhub builds

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -422,31 +403,40 @@ on-integ-and-version-tags: &on-integ-and-version-tags
...
           <<: *on-any-branch
-      - build-debian:
+      - build-and-test:
           <<: *on-any-branch
           <<: *after-linter
+      - dockerhub-build:
+          name: dockerhub-edge
+          build_args: EDGE=1
+          <<: *after-build-and-test
+          <<: *on-any-branch
```
Comment (DvirDukhan) [inline .circleci/config.yml:439]:
```suggestion
          <<: *on-master
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -106,6 +106,29 @@ commands:
...
+      - checkout
+      - run:
+          name: Setup
+          command: |
+            git submodule update --init --recursive
+            ./opt/readies/bin/getpy3
+      - run:
+          name: Build for platform
+          command: |
+            #@@ docker login -u redisai -p $DOCKER_REDISAI_PWD
```
Comment (DvirDukhan) [inline .circleci/config.yml:124]: comment?

File: .circleci/config.yml [hunk 3]
```diff
@@ -106,6 +106,29 @@ commands:
...
+            cd opt/build/docker
+            #@@ make build publish $(./version-params) CPU=1 OSNICK=bionic X64=1 REPO=redisai VERBOSE=1 <<parameters.build_args>>
+            #@@ make build publish $(./version-params) GPU=1 OSNICK=bionic X64=1 REPO=redisai VERBOSE=1 <<parameters.build_args>>
```
Comment (DvirDukhan) [inline .circleci/config.yml:127]: comment?

**Review by DvirDukhan — COMMENTED** [review]
thanks @rafie !
1. edge is only on master
2. where is the logic between tagging a `1.0.x` to `1.2.x` with respect to the `latest` tag?

File: .circleci/config.yml (hunk 1 above)
Comment (rafie) [reply .circleci/config.yml:439]: (1) this is for testing the PR. will change to "on-master" once it is working.
(2) true. will add support for `LATEST_BRANCH=1.2`

File: .circleci/config.yml (hunk 3 above)
Comment (rafie) [reply .circleci/config.yml:127]: Don't yet have password for `dockerhub/redisai` so publish will fail.

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**